### PR TITLE
Other command for SMTP nio library

### PR DESCRIPTION
**Pull request by fansu**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Major release (change is NOT backward compatible with prior release)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by fansu** [comment]
@OnePunMan
Hi Jackie, here are your other commands. Feel free to contribute if you are interested.

**Comment by OnePunMan** [comment]
Thanks @fansu, I'll start looking into them

**Comment by rockwotj** [comment]
@fansu is this PR going to be merged?

File: core/src/main/java/com/yahoo/smtpnio/async/request/DataCommand.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        super(DATA);
+        this.message = message;
+    }
+
+    @Override
+    public ByteBuf getNextCommandLineAfterContinuation(@Nonnull final SmtpResponse serverResponse) {
+        try {
+            final ByteArrayOutputStream output = new ByteArrayOutputStream();
+            message.writeTo(output);
+            return Unpooled.buffer(output.size() + SmtpClientConstants.PADDING_LEN)
```
Comment (chibenwa) [inline core/src/main/java/com/yahoo/smtpnio/async/request/DataCommand.java:46]: The SMTP command abstraction would, on the long run, deserved to be rework in order not to get the full representation of the email in-memory...

File: core/src/main/java/com/yahoo/smtpnio/async/request/DataCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ * This class defines a Data (DATA) command.
+ */
+public class DataCommand extends AbstractSmtpCommand {
+
+    /** String literal for "DATA". */
+    private static final String DATA = "DATA";
+
+    /** The message to be sent. */
+    private SMTPMessage message;
```
Comment (chibenwa) [inline core/src/main/java/com/yahoo/smtpnio/async/request/DataCommand.java:29]: Forcing the usage of Javax Mail seems unecessary to me. 

Why not passing an agnostic representation of `byte[]` ?

(In a reactive world, `Publisher<ButeBuffer>`)

**Review by chibenwa — COMMENTED** [review]
Sounds very interesting indeed.

What is the ETA of this work?

**Comment by chibenwa** [comment]
Is a refresher of this PR worth my time?

**Comment by fansu** [comment]
> Is a refresher of this PR worth my time?

Thanks for your attention and feel free to contribute.

**Comment by fansu** [comment]
Closed. PR 42 merged with duplicate changes.

**Pull request closed without merging**